File: 04-CSHARP/08-Funciones/06pasoPorReferenciaArrays.cs
// Paso por referencia de arrays en C#

void CambiarValor(int[] parametro)
{
    parametro[0] = 20;
}

// Los arrays se pasan por referencia por defecto en C#
int[] argumento = {10};

Console.WriteLine($"Valor antes de la funcion: {argumento[0]}");
CambiarValor(argumento);
Console.WriteLine($"Valor despues de la funcion: {argumento[0]}");

File: 02_C/08_funciones/06pasoPorReferenciaArrays.c
#include <stdio.h>

void cambiarValor(int parametro[])
{
    // Cambia los valores del array desde la funcion.
    parametro[0] = 4;
    parametro[1] = 5;
    parametro[2] = 6;
}

char *cambiarValorChar(char *parametro)
{
    parametro = "Adios";

    return parametro;
}

int main()
{
    // Ejemplo de paso por referencia: Los arrays y matrices siempre se pasan por referencia.
    int arg[] = {1, 2, 3};

    printf("Valor antes de funcion: %d %d %d\n", arg[0], arg[1], arg[2]);
    cambiarValor(arg);
    printf("Valor despues de funcion: %d %d %d\n", arg[0], arg[1], arg[2]);

    // Paso por referencia con cadenas
    // char arg2[] = "Hola a todos";
    char *arg2 = "Hola a todos";
    printf("\nValor antes de funcion: %s\n", arg2);
    arg2 = cambiarValorChar(arg2);
    printf("Valor despued de funcion: %s", arg2);

    return 0;
}